### ISSUE TITLE
Add *.github.io to connect-src CSP

### ISSUE THREAD

**Issue by imax9000** [DestinyItemManager/DIM#9500]
### Proposed change

Can you please add `https://*.github.io` to `connect-src` list in the `Content-Security-Policy` header?

### How does this fit into your workflow?

This would allow the wishlists generated by Github Actions ([example](https://github.com/imax9000/dim-wish-list-sources/blob/automation/.github/workflows/strip.yml)) to be published [directly to Github Pages](https://github.com/imax9000/dim-wish-list-sources/blob/automation/.github/workflows/pages.yml) w/o polluting the repository with generated files.

**Comment by robojumper** [comment]
This was previously discussed on the Discord [here](https://discord.com/channels/316217202766512130/333464468241514497/984223942585761803) with the objection that "polluting the repository" to load the generated lists from `raw.githubusercontent.com` isn't that bad.

**Comment by imax9000** [comment]
That isn't really an objection. What's the cost/risk of adding this entry?

**Comment by bhollis** [comment]
We like the fact that `raw.githubusercontent.com` is a single domain (vs. a wildcard) and that the `raw` server is specifically set up to serve pretty much plain text without any fancy headers or anything. In contrast, GitHub pages gives a fair amount of control and can be arbitrary stuff. That's not to say that the risk is that much greater, but we'd prefer to keep our CSP as locked down as we practically can.

**Comment by imax9000** [comment]
That makes some sense, thanks.

FWIW, Pages are quite locked down too, only static content can be served
from there, no fancy headers or anything.

And since we're taking only about `connect-src`, browser won't ever attempt
to interpret the response as markup or code. So "arbitrary stuff" boils
down only to returning connect with an appropriate Content-Type instead of
forcing text/plain, but that makes no difference for both XMLHttpRequest
and fetch() APIs.

**Comment by bhollis** [comment]
Agreed - as I said, the risk profile isn't that different. It's mostly about keeping the allowed list of domains small.
